packages/celotool/src/lib/geth.ts

```diff
@@ -520,26 +575,67 @@ export const simulateClient = async (
     // randomly choose which gas currency to use
     const feeCurrencyGold = Boolean(Math.round(Math.random()))
 
-    let feeCurrency
-    if (!feeCurrencyGold) {
-      try {
-        feeCurrency = await kit.registry.addressFor(CeloContract.StableToken)
-      } catch (error) {
-        tracerLog({
-          tag: LOG_TAG_CONTRACT_ADDRESS_ERROR,
-          error: error.toString(),
-          ...baseLogMessage,
-        })
+    let feeCurrency, txOptions
+    try {
+      feeCurrency = feeCurrencyGold ? '' : await kit.registry.addressFor(CeloContract.StableToken)
+    } catch (error) {
+      tracerLog({
+        tag: LOG_TAG_CONTRACT_ADDRESS_ERROR,
+        error: error.toString(),
+        ...baseLogMessage,
+      })
+    }
+
+    try {
+      feeCurrency = feeCurrency || ''
+      baseLogMessage.feeCurrency = feeCurrency
+
+      const gasPriceMinimum = await kit.contracts.getGasPriceMinimum()
+
+      const gasPriceBase = feeCurrency
+        ? await gasPriceMinimum.getGasPriceMinimum(feeCurrency)
+        : await gasPriceMinimum.gasPriceMinimum()
+      let gasPrice = new BigNumber(gasPriceBase).times(2)
+
+      // Check if last tx was mined. If not, reuse the same nonce
+      if (lastTx === '' || lastNonce === 0) {
+        nonce = await kit.connection.nonce(kit.defaultAccount)
+      } else if ((await kit.connection.getTransactionReceipt(lastTx))?.blockNumber) {
+        nonce = await kit.connection.nonce(kit.defaultAccount)
+      } else {
+        nonce = (await kit.connection.nonce(kit.defaultAccount)) - 1
+        gasPrice = gasPrice.times(6)
```

Comment:
I would prefer a more adaptive approach in case we need to replace with the same nonce multiple times, as we discussed (keep track of the last used gas price with current nonce and if we need to replace then bump it up by 15%)